pkg/sync/service.go

```
@@ -232,6 +237,66 @@ func (s *Service) DownloadToDst(writer http.ResponseWriter, request *http.Reques
+	}
+	sf.lock.RUnlock()
+
+	// If it is for v2 creation, we don't compress the data and will temporarily transform the qcow2 to raw image
+	if forV2Creation == "true" {
+		downloadFile := sf.filePath
+		tmpRawFile := ""
+
+		// If the source file is qcow2 when encrypting we need to convert and use its raw image
+		imgInfo, err := util.GetQemuImgInfo(downloadFile)
+		if err != nil {
+			err = fmt.Errorf("failed to detect if file is qcow2 %v", downloadFile)
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)